docs/source/conf.py

```diff
@@ -90,13 +91,34 @@
 }
 
 
+def autodoc_process_signature(
+    app, what, name, obj, options, signature, return_annotation
+):
+    """Modify found signatures to fix various issues."""
+    if signature is not None:
+        if name.startswith("trio.testing"):
+            # Expand type aliases
+            signature = re.sub(
+                r"StreamMaker\[([a-zA-Z ,]+)]",
+                lambda match: f"typing.Callable[[], typing.Awaitable[tuple[{match.group(1)}]]]",
+                signature,
+            )
+            signature = signature.replace(
+                "AsyncHook", "typing.Callable[[], typing.Awaitable[object]]"
+            )
+            signature = signature.replace("SyncHook", "typing.Callable[[], object]")
+
+    return signature, return_annotation
```

Comment:
Oh wow, having to do this whenever we introduce type aliases in public functions is a major pain. Thankfully not too bad to expand this function, but :<

Reply:
There's probably a better approach, we could parse with the AST module, introspect the function object etc. Though the advantage is we can do whatever we want to the signature here.

The alternative is to publicly document type aliases, might need a custom directive to properly add them as a "class"...

Reply:
some of the type aliases might make sense to publicly document, as the same logic of "make the signature readable" and more specifically "make it obvious that all these functions take the same type of parameter" is valid just as much, if not more, in reading the docs as when reading the code.

And yeah, looking at https://trio--2747.org.readthedocs.build/en/2747/reference-testing.html#trio.testing.MemorySendStream it's ... far from readable.

(just realized: would be great if readthedocs provided a diff view)

Reply:
Sphinx 7.0 has the functionality to automatically swap to 1 arg per line for long signatures. Once the [next version of `sphinx_rtd_theme`](https://github.com/readthedocs/sphinx_rtd_theme/pull/1464) releases, we should be able to use that.